packages/client/src/generation/generateClient.ts

```diff
@@ -125,6 +125,16 @@ export async function buildClient({
   if (dataProxy === true) {
     fileMap['edge.js'] = await JS(edgeTsClient, true)
     fileMap['edge.d.ts'] = await TS(edgeTsClient, true)
+
+    // Generates shims for deno
+    fileMap['deno.ts'] = `import "https://deno.land/std@0.148.0/dotenv/load.ts";
```

Comment:
How does this work? Does it mean you need to import the generated client like `import ... from '.prisma/client/deno.ts'` ?
If that is the case, we can simplify things by not touching `edge` related stuff and have a dedicated deno entry point.

Reply:
(Related: I'm also thinking that one day we might have support for N-API (Query Engine) *and* Deno Deploy (Data Proxy), so it might make sense to plan have different entry points from the start).

Reply:
> How does this work? Does it mean you need to import the generated client like import ... from '.prisma/client/deno.ts' ?

Yes.

> If that is the case, we can simplify things by not touching edge related stuff and have a dedicated deno entry point.

My previous PR does that and didn't change anything in edge.js (See https://github.com/prisma/prisma/pull/14250). I got feedback from @matthewmueller that suggests unifying edge.js and deno.ts. So I modified edge.js and made it work on both Deno and Cloudflare (I admit it's a little bit trickier solution than the previous one). I'm open to both directions.

Reply:
If we want to keep a single entry point for `edge`, we could as well do this dynamically the the generated client. Check that `Deno` exists and then polyfill the `process` with `Deno.env` if that is the case.

Reply:
Can `import "https://deno.land/std@0.148.0/dotenv/load.ts"` be made dynamic with top-level await?